README.md

```diff
@@ -11,9 +11,92 @@ multi-repo syncronization &amp; management tool
+
+For you to run leif, you need the `leif` CLI (this repo), a `leif.yml` file and a GitHub API token with read/write access to your repos. The `leif.yml` file (see the [example](#example-leif.yml) below) is where you list GitHub repos names and define assertions. An assertion is explicit state about a repo. "State", as used here, means directories, files and GitHub repo properties. When run, the `leif` CLI will read the `leif.yml` file and check that each listed repo has the defined state. If not, leif creates a PR to the repo with the applicable changes or sets the GitHub repo property via the GitHub API.
+
+_Leif asserts explicit state on repos_.
+
+# Definitions
+**assertion** - A user defined state of a repo. Users define assertions using of the many [Assertion types](#assertions). If the repo's state differs from the assertion, leif commits the changes; an assertion becomes a single commit.
+
+**sequence** - A logical grouping of one or more assertions. Leif allows us to chain many assertions into a sequenece and run them all syncronously. `leif.yml` can have one or many sequences. A sequence becomes a GitHub PR.
+
+**workflow** - A workflow marries repos with sequences. With workflows, users define which sequences run on which repos. `leif.yml` can have one or many workflows.
+
+**repo groups** - A logical grouping of repos in the same GitHub org, ex: core repos, documentation repos, forked repos, etc.
+
+# Example leif.yml
+```yaml
+---
+repos:
+  - rasphilco/leif
+  - group: foo-example-org
+    github_org: foo-example
+    repos:
+      - bar
+      - baz
+sequences:
+  sync-package.json:
+    description: 'sync pjson properties'
+    assertions:
+      - type: json-has-properties
+        description: 'updating properties'
+        source_relative_filepath: state/package.json
+        target_relative_filepath: package.json
+  sync-dependabot:
+    description: 'sync dependabot.yml'
+    assertions:
+      - type: file-is-exact
+        description: 'updating dependabot'
+        source_relative_filepath: state/dependabot.yml
+        target_relative_filepath: .github/dependabot.yml
+
+workflows:
+  sync-repo:
+    apply_to_repos:
+      - rasphilco/leif
+    apply_to_groups:
+      - foo-example-org
+    sequences:
+      - test-assertions
+      - sync-dependabot
+```
+
+# Assertions
+
+* [file-does-not-exist](#type-file-does-not-exist)
+* [file-is-exact](#type-file-is-exact)
+
+#### type: file-does-not-exist
+
+Property | Example | Description
+--- | --- | ---
+description | `remove old LICENSE` | describe what is transpiring, which will become the commit message
+if | `test -e "./new/LISCENSE.txt"` | assertion only runs if this bash expression returns successfully
```

Comment:
Do `if`s have to be bash expressions?

Reply:
Currently, yes. Could make it run a shell script.